corehq/apps/app_manager/suite_xml/sections/remote_requests.py

```diff
@@ -90,14 +90,14 @@ def _build_instances(self):
             if prop.itemset.instance_id
         ]
 
-        query_xpaths = [QuerySessionXPath(self.module.search_config.case_session_var).instance()]
-        query_xpaths.extend([datum.ref for datum in self._get_remote_request_query_datums()])
-        query_xpaths.extend([self.module.search_config.get_relevant(), self.module.search_config.search_filter])
-        query_xpaths.extend([prop.default_value for prop in self.module.search_config.properties])
-        instances, unknown_instances = get_all_instances_referenced_in_xpaths(self.app, query_xpaths)
+        xpaths = [QuerySessionXPath(self.module.search_config.case_session_var).instance()]
```

Comment:
...I now see you added a convert to a set in `get_all_instances_referenced_in_xpaths`, but I think you could use sets everywhere and it would simplify the code a little bit.

Reply:
good call - there was definitely some room for simplification: https://github.com/dimagi/commcare-hq/pull/30239/commits/91bd644e42e485ada84e68cc35c9ed3edb606906